### PR TITLE
yum: Parse also obsoleted packages

### LINKED ISSUE #39978: yum: Don't list obsoleted packages
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

yum

##### ANSIBLE VERSION

```
2.5.2
```

##### SUMMARY

When running `yum` module with `state=latest` and `name=*` the yum module don't report obsoleted packages.

##### STEPS TO REPRODUCE

Have a host which has packages which obsoletes some package:

```yaml
- name: Update system
  yum:
    name: *
    state: latest
```

```bash
[root@host ~]# yum check-update
Obsoleting Packages
package.noarch                                                                 4.2-20180508.0.el7_5                                                  repo
    package-placeholder.noarch
```

##### EXPECTED RESULTS

Packages should be listed return of the `yum` module, for example in `obsoleted` list or as part of updated.

##### ACTUAL RESULTS

The packages are not listed.

```

```
The issue is here:

https://github.com/ansible/ansible/blob/2de6734ce49cdc54b2e01e720a0229da5defbbf4/lib/ansible/modules/packaging/os/yum.py#L1054

The module checks if line has 3 parts, but in this case we have 6 parts.

### PR DESCRIPTION
**Pull request by machacekondra**
##### SUMMARY

Fixes: https://github.com/ansible/ansible/issues/39978

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

yum

##### ANSIBLE VERSION

```
2.5.2
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/packaging/os/yum.py [hunk 1]
```diff
@@ -1051,10 +1051,10 @@ def parse_check_update(check_update_output):
         #
         # FIXME: what is  the '.' not in line  conditional for?
 
-        if '*' in line or len(line) != 3 or '.' not in line[0]:
+        if '*' in line or len(line) not in [3, 6] or '.' not in line[0]:
```
Comment (mkrizek) [inline lib/ansible/modules/packaging/os/yum.py:1056]: Can you also mention what the 6 means in the comment above, please?

**Review by mkrizek — CHANGES_REQUESTED** [review]
Unit tests failed, looks like tests for obsolete packages are already in `test/units/modules/packaging/os/test_yum.py`.

**Comment by machacekondra** [comment]
The unit test says that the obsoleted packages shouldn't be part of the updated packages. But we need to handle those somehow, as it's valid use case there is some package obsoloting another we need to perform system update.

**Comment by mwperina** [comment]
> The unit test says that the obsoleted packages shouldn't be part of the updated packages. But we need to handle those somehow, as it's valid use case there is some package obsoloting another we need to perform system update.

Ondro could you fix (or remove if not possible) those tests? We need to handle this valid use case

@machacekondra

**Comment by maxamillion** [comment]
@machacekondra any update here?

**Pull request closed without merging**